csharp/ql/src/semmle/code/csharp/ir/implementation/raw/internal/InstructionTag.qll

```diff
@@ -0,0 +1,169 @@
+import csharp
+import semmle.code.csharp.ir.Util
+
+
+// TODO: Both exists needed?
+private predicate fieldIsInitialized(Field field) {
+  exists(ObjectInitializer initList | 
+  	exists(int i | field = initList.getMemberInitializer(i).getInitializedMember())
+  ) or
+  exists(MemberInitializer init |
+    field = init.getInitializedMember()
+  )
+}
+
+private predicate elementIsInitialized(int elementIndex) {
+  exists(ArrayInitWithMod initList |
+    initList.isInitialized(elementIndex)
+  )
+}
+
+newtype TInstructionTag =
```

Comment:
I would hope that this file could be shared.

Reply:
These tags are specific to how each language is translated to IR. Obviously, there's a lot of overlap between C# and C++ for things like arithmetic operations and assignment. I think starting with a copy-paste from C++ and diverging from there is OK. There's no need to keep these files in sync between languages.